### PR TITLE
Status - System Logs - VPN

### PR DESCRIPTION
**Pull request by NOYB**
Revamp log file selection method to be consistent with the other system logs.

Add advanced filter section; only activated for the non-vpn.log files (needs testing with vpn.log file first, continue using dump_clog_vpn until tested).

Add manage log section

### CODE REVIEW

File: src/etc/inc/filter_log.inc [hunk 1]
```diff
@@ -87,9 +87,10 @@ function conv_log_filter($logfile, $nentries, $tail = 50, $filtertext = "", $fil
 	else if ($logfile == "{$g['varlog_path']}/ipsec.log")		{ $logfile_type = "system"; }
 	else if ($logfile == "{$g['varlog_path']}/ppp.log")			{ $logfile_type = "system"; }
 
-	else if ($logfile == "{$g['varlog_path']}/vpn.log")			{ $logfile_type = "tbd"; }
-	else if ($logfile == "{$g['varlog_path']}/poes.log")		{ $logfile_type = "tbd"; }
-	else if ($logfile == "{$g['varlog_path']}/l2tps.log")		{ $logfile_type = "tbd"; }
+# Needs to be tested with a vpn.log file before enabling.
```
Comment (sbeaver-netgate) [inline src/etc/inc/filter_log.inc]: Please email me and I will reply with some sample vpn logs for you to test with.

**Comment by NOYB** [comment]
Steve,
Sent an email to you yesterday re: sample vpn log file.  Know your real busy, so just wanted to check that you received it.

**Comment by sbeaver-netgate** [comment]
Sorry for the delay. Attached is my vp.log from a system running IPSec.

Please take care not to share anything in this file, and delete it once you are done.

Steve
--
Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com | www.netgate.com

**Comment by rbgarga** [comment]
@sbeaver-netgate can you check this please?

**Comment by sbeaver-netgate** [comment]
Since we have not been able to come up with a usable vpn log file for testing, I suggest we merge this and ask the community for help testing via a note on the forum.

**Comment by NOYB** [comment]
In what state should it be merged for community testing?  As-is using the dump clog vpn function (no filtering for vpn.log)?  Or using the advanced filter?  Bear in mind the regex for the advanced filter is most certainly not correct.  I just took a wild swing at reverse engineering it from the dump clog vpn code.

This only applies to the vpn.log file. I think the l2tps.log, and poes.log should be okay as I was able to generate at least basic logs for those.

**Comment by sbeaver-netgate** [comment]
Consulted JimP on this. It looks like vpn.log is used only to record login/logout of PPPoE and L2TP. The code is in /usr/local/sbin/vpn-link-up (From which pptp should really have been removed)

If you set up one of those two and log in/out a couple of times, you should generate some logs.

Still don't understand why the zipped logs I sent you didn't work.  Perhaps we should compare checksums or something?

**Comment by sbeaver-netgate** [comment]
The logs should look something like:

Jan  6 09:25:10 hostname root: login,poes0,192.168.2.135,jim
Jan  6 09:25:20 hostname root: logout,poes0,192.168.2.135,jim

**Comment by sbeaver-netgate** [comment]
Try this from the command line:
````bash
/usr/bin/logger -p local3.info "login,poes10,192.168.8.8,someuser"
/usr/bin/logger -p local3.info "logout,poes10,192.168.8.8,someuser"
````
Then navigate to: Status->system Logs->vpn->pppoe Login abd you will see the login/out event.

I also see "No logs to display", so there is an issue there :)

**Comment by sbeaver-netgate** [comment]
Looks like this is ready to go now. Would you confirm please and I will test and merge?

**Comment by NOYB** [comment]
Yes, I think it's ready.

**Comment by sbeaver-netgate** [comment]
Great. Thanks for all the work you have done on this one. I'll start testing now.

**Pull request merged**